File: extract.py
import sys

def extract_schema(input_file, output_file):
    with open(input_file, 'r', encoding='utf-8', errors='ignore') as f_in, \
         open(output_file, 'w', encoding='utf-8') as f_out:
        
        in_create = False
        for line in f_in:
            if line.startswith('CREATE TABLE'):
                in_create = True
            
            if in_create:
                f_out.write(line)
                if line.strip().endswith(';'):
                    in_create = False
                    f_out.write('\n')
                    continue
            
            if line.startswith('ALTER TABLE') or line.startswith('DROP TABLE'):
                f_out.write(line)

if __name__ == '__main__':
    extract_schema(sys.argv[1], sys.argv[2])

File: extract.php
<?php

$inFile = $argv[1];
$outFile = $argv[2];

$in = fopen($inFile, "r");
$out = fopen($outFile, "w");

$in_create = false;
while (($line = fgets($in)) !== false) {
    if (strpos($line, "CREATE TABLE") === 0) {
        $in_create = true;
    }
    
    if ($in_create) {
        fwrite($out, $line);
        if (substr(trim($line), -1) === ';') {
            $in_create = false;
            fwrite($out, "\n");
        }
        continue;
    }
    
    if (strpos($line, "ALTER TABLE") === 0 || strpos($line, "DROP TABLE") === 0) {
        fwrite($out, $line);
    }
}

fclose($in);
fclose($out);
